storage version integration test: check the test server's health before running

roycaihw opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide/first-contribution.md#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
/kind flake

**What this PR does / why we need it**:
Previously we disabled the `/healthz` check completely because the test blocks one post-start hook from finishing. As a result we risk sending requests to the server before it's ready. Instead we should check all the `/healthz/...` endpoints except the blocked one before running the test. 

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

/assign @liggitt @caesarxuchao
/sig api-machinery

liggitt commented on test/integration/storageversion/storage_version_filter_test.go:
```diff
@@ -253,3 +264,40 @@ func TestStorageVersionBootstrap(t *testing.T) {
+		t.Fatalf("error parsing root paths: %v", err)
+	}
+
+	// Check all healthz endpoints except built-in-resources-storage-version-updater
```
can you use `/healthz?exclude=built-in-resources-storage-version-updater`?

roycaihw replied:
thanks for the pointer. I will update the PR

liggitt commented on test/integration/storageversion/storage_version_filter_test.go:
```diff
@@ -253,3 +264,40 @@ func TestStorageVersionBootstrap(t *testing.T) {
+			continue
+		}
+		t.Run(fmt.Sprintf("healthz endpoint %s", path), func(t *testing.T) {
+			if err := wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (bool, error) {
```
is 10 seconds long enough? how long does the normal integration test wait for /healthz?

roycaihw replied:
it uses one minute. I will change to that as we only need to query one endpoint now

roycaihw committed:
storage version integration test: check the test server's health before running

we disabled the /healthz check because our test blocks one post-start
hook from finishing. Instead we should check all the other /healthz/...
endpoints before running the tests

roycaihw commented:
@liggitt Updated. Most of the changes now are whitespace changes. PTAL

liggitt commented:
/lgtm

liggitt commented:
/approve

jpbetz commented:
/triage accepted
